frontend/src/container/FormAlertChannels/index.tsx

```diff
@@ -52,14 +65,13 @@ function FormAlertChannels({
 						<Option value="slack" key="slack">
 							Slack
 						</Option>
+						<Option value="webhook" key="webhook">
+							Webhook
+						</Option>
 					</Select>
 				</FormItem>
 
-				<FormItem>
-					{type === 'slack' && (
-						<SlackSettings setSelectedConfig={setSelectedConfig} />
-					)}
-				</FormItem>
+				<FormItem>{renderSettings()}</FormItem>
```

Comment:
can you please update this function 

by 

condition && <ComponentA>
conditionB && <ComponentB>

Reply:
as switch with default case is required here

Reply:
Do you want to remove switch? and replace with && conditions (like below). 

type=== 'slack' && <SlackSettings setSelectedConfig={setSelectedConfig} />

type === 'webhook && <WebhookSettings setSelectedConfig={setSelectedConfig} /> 

The reason I separated the function and switch is we have more setting options coming in next releases.   

Please share more details on your suggested approach.

Reply:
you can use switch no issue ... but if you want use that please throw error while there is no default component

Reply:
Changed this. Returning NULL for default case. As such a rare scenario since type is mandatory in the form, but in case it is not in the future, the settings will be displayed only when type is selected.

I thought not showing settings is a valid situation here rather than throwing error.  When user selects a type, the correct settings will be displayed. 

Hope this makes sense.